design/authenticators/application-identity/application-identity.md

```diff
@@ -0,0 +1,464 @@
+The Application Identity gives the authenticator another layer of
+authentication, where it can authenticate applications using not only their
+Conjur identity and permissions (e.g host can `authenticate` with the
+authenticator `webservice`) but also their native identity (e.g Azure
+resource characteristics)
+ 
+ For example, in the Azure Authenticator the Application Identity
+  includes:
+   - Subscription ID
+   - Resource Group
+   - System-Assigned Identity (optional)
+   - User-Assigned Identity (optional)
+ 
+A host that will authenticate using the Azure authenticator will be defined
+ as follows:
+ ```
+- !host 
+  id: test-app 
+  annotations: 
+    authn-azure/subscription-id: test-subscription 
+    authn-azure/resource-group: test-group 
+``` 
+
+In the example above, the Application Identity is stored as annotations.
+In the Kubernetes Authenticator the Application Identity can be stored either
+in the host's ID or in its annotations. 
+For example:
+ ```
+- !host
+  id: test-app-service-account
+  annotations:
+    authn-k8s/namespace: some-namespace
+    authn-k8s/service-account: some-service-account
+
+- !host
+  id: some-namespace/service_account/some-service-account
+```
+
+In the example above, both hosts have the same Application Identity. One
+ stores it in its annotation and the other in its ID.
+
+## Issue Description
+
+When we add a new authenticator to Conjur, we need to add the mechanism
+of validating the host's Application Identity. Currently we have no generic
+way to do it and because of that it is less straight-forward to add a new
+authenticator. An example of what a generic implementation is can be seen in
+the [Authenticator Status Check](../authenticators-status). As you can see
+in the [Implementing Authenticator Status API
+](../authenticators-status/authn-status-new-implem.md) doc, it is very simple
+to add a new Status Check to an Authenticator. This happens because the flow
+starts with general checks that are relevant to all authenticators, and all
+that needs to be done in order to add a new Status Check is to implement a
+method named `status` in the relevant `Authenticator` class, and create a
+ Command Class that will perform the actual validations. 
+ 
+However, when validating the Application Identity we currently have no
+generic mechanism, as can be seen in the implementations for the [Azure
+Authenticator](/app/domain/authentication/authn_azure/validate_application_identity.rb) and 
+[Kubernetes Authenticator](/app/domain/authentication/authn_k8s/validate_application_identity.rb), where a lot of code is duplicated with
+ minor changes. 
+
+## Solution
+
+Before we can find the solution, we need to understand how Application
+ Identities are related across authenticators, and how they are different.
+ 
+At this point, to ease the problem, I will act as if Application Identities
+can be defined only in the host's annotations, and not in the ID. In future
```

Comment:
For the sake of simplicity, from here and throughout the doc, Application Identities will be defined only in the host's annotations, and not in the ID.